### PR TITLE
Fixes #15783: Make class_prefix unique

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/15783

### CODE REVIEW

File: tree/30_generic_methods/command_execution.cf [hunk 1]
```diff
@@ -21,27 +21,29 @@
...
-# 
+#
 # @class_prefix command_execution
 # @class_parameter command_name
 
 bundle agent command_execution(command_name)
 {
   vars:
-      "old_class_prefix"  string => canonify("command_execution_${command_name}");
+      "old_class_prefix"  string => canonify("${report_data.canonified_directive_id}_command_execution_${command_name}");
```
Comment (Fdall) [inline tree/30_generic_methods/command_execution.cf]: This is tricky we are changing the expected outcome classes, maybe we should add a third one ?

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR rebased

File: tree/20_cfe_basics/ncf_lib.cf [hunk 2]
```diff
@@ -1173,7 +1173,7 @@ bundle agent ncf_services(service, action)
       "canonified_action"         string => canonify("${action}");
       "canonified_action_command" string => canonify("${action_command}");
 
-      "class_prefix"  string => "ncf_services_${canonified_service}_${canonified_action}";
+      "class_prefix"  string => "${report_data.canonified_directive_id}_ncf_services_${canonified_service}_${canonified_action}";
```
Comment (ncharles) [inline tree/20_cfe_basics/ncf_lib.cf]: This breaks all services methods, that use "ncf_services_*" classes for all conditions

**Comment by Fdall** [comment]
This not good enough, we should wait to have a proper id for each call done by the generation before trying to fix this, or it will seem super hacky

**Pull request closed without merging**